### PR TITLE
Conduits Automated Mover

### PR DESCRIPTION
**Pull request by oldchili**

### CODE REVIEW

File: test/funnels/automation/ConduitMover.t.sol [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        buffer = address(new AllocatorBuffer());
+        AllocatorRoles roles = new AllocatorRoles();
+        AllocatorRegistry registry = new AllocatorRegistry();
+        registry.file(ILK, "buffer", buffer);
+
+        conduit1 = address(new AllocatorConduitMock(address(roles), address(registry)));
+        conduit2 = address(new AllocatorConduitMock(address(roles), address(registry)));
+        mover    = new ConduitMover(ILK, buffer);
+
+        // Allow mover to to perform ILK operations on the conduits
```
Comment (telome) [inline test/funnels/automation/ConduitMover.t.sol]:
```suggestion
        // Allow mover to perform ILK operations on the conduits
```

**Review by telome — APPROVED** [review]
Minor typo, the rest looks good!

File: src/funnels/automation/ConduitMover.sol [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        });
+        emit SetConfig(from, to, gem, num, hop, lot);
+    }
+
+    function move(address from, address to, address gem) toll external {
+        MoveConfig memory cfg = configs[from][to][gem];
+
+        require(cfg.num > 0, "ConduitMover/exceeds-num");
+        require(block.timestamp >= cfg.zzz + cfg.hop, "ConduitMover/too-soon");
+        configs[from][to][gem].num = cfg.num - 1;
```
Comment (sunbreak1211) [inline src/funnels/automation/ConduitMover.sol:99]: This can be unchecked. We haven't done it for the Stable Swapper and Depositor though.

**Review by sunbreak1211 — APPROVED** [review]
Same, minor comment but rest looks good.

File: src/funnels/automation/ConduitMover.sol (hunk 2 above)
Comment (oldchili) [reply src/funnels/automation/ConduitMover.sol:99]: I'll leave it as is and we can do it for all together later if we think it's worthy.

**Pull request merged**